### ISSUE TITLE
MQTT Configuration in  /etc/thingsboard/conf/thingsboard.yml - What it is for?

### ISSUE THREAD

**Issue by ChSM56** [thingsboard/thingsboard#483]
**I am still to Understand the below Configuration** 
****************************************************
Ref:  /etc/thingsboard/conf/thingsboard.yml

I have mosquitto mqtt service running on port 1883. So I had to change the mqtt section with port to 1882 whereas my understanding was that it should be 1883 as the platform is to connect to the broker but a service runs on this port and I am not sure what it does?.

****************************************************
![image](https://user-images.githubusercontent.com/29800826/34036396-1640ec5a-e1ab-11e7-8c7e-6e6fdac44d86.png)

**Comment by zhw618** [comment]
when  you change .yml file from port 1883 to 1882，
the thingsboard will start a embedded mqtt broker using port 1882.

**Comment by ChSM56** [comment]
@zhw618 👍 

If I do not run mosquitto (actually I switched to HiveMq courtesy Sitewhere platform uses it and I wanted to minimize the brokers running on the system ; running both on my system), I can not see the complete message flow happening.

Scenario 1: No 3rd party Broker and TB_Gateway starts the broker as @zhw618  said above expecting all mqtt messages must flow back and forth. Not really happening. I am using 2 systems - 1 Centos 7 running Thingsboard platform and Windows 10 desktop (mosquitto).

Scenario 2: Centos 7 with NO TB_Gateway running HiveMQ broker and Windows 10 TB_Gateway as well as Raspberry Pi 3 . Neither of the 2 TB_Gateways connect to the Platform and I can not see the flow happening.

Basically, I am exploring to configure and deploy multiple TB_Gateways based on local conditions and the Platform to process the incoming data.

Thank you, so much.


Please help me what should be the configurations on these cases? Assume any internal IP numbers and ports on these systems.

I am one of those that would first explore the software architecture and components for the degree of freedom and inter connected by loose coupling as a high priority.

**Comment by ChSM56** [comment]
Here is the picture showing what I am attempting very seriously to make work and the ONLY set-up I could make to work is ALL in ONE System with Centos 7.

![thingsboard config](https://user-images.githubusercontent.com/29800826/34389319-ee6ea290-eb5e-11e7-88cb-7352aaf67024.png)

As I read in one of the Google Groups post, I made an attempt to build from Source and reached a dead end.

![thingboard v1 4 build failure screen](https://user-images.githubusercontent.com/29800826/34389331-0dab2d86-eb5f-11e7-8087-da08b2859365.PNG)

Thank you for all the help.

**Comment by ChSM56** [comment]
[centos7-mqtt-config.json.txt](https://github.com/thingsboard/thingsboard/files/1609662/centos7-mqtt-config.json.txt)
[centos7-tb-gateway.yml.txt](https://github.com/thingsboard/thingsboard/files/1609663/centos7-tb-gateway.yml.txt)
[centos7-thingsboard.yml.txt](https://github.com/thingsboard/thingsboard/files/1609664/centos7-thingsboard.yml.txt)


I could make the landscape work for me with 2 instances of tb-gateway running on CENTOS 7 and WINDOWS 10 System with mqtt broker running on CENTOS 7.

https://user-images.githubusercontent.com/29800826/34389319-ee6ea290-eb5e-11e7-88cb-7352aaf67024.png

The working config files are attached - 
[ws10-mqtt-config.json.txt](https://github.com/thingsboard/thingsboard/files/1609659/ws10-mqtt-config.json.txt)
[ws10-tb-gateway.yml.txt](https://github.com/thingsboard/thingsboard/files/1609660/ws10-tb-gateway.yml.txt)


When I move around the MQTT broker to either WS10 system or raspberry running mosquitto, the configurations does not work.

I am still curious on how to move MQTT broker away from CENTOS 7 System.

Any help is much much appreciated.

**Comment by ChSM56** [comment]
Yes, I could make the above landscape work with MQTT Broker running on either Windows 10 or Raspberry with ease of configuration changes.

Further, I established with 2 Gateways, it does not matter which Gateway a Device is connecting. The KEY is in the file - "mqtt-config.json" the IP should point to MQTT Broker System.

The mistake that I made often is to change the IP in both the files to be the same.
![mqtt-config json](https://user-images.githubusercontent.com/29800826/34649458-d18cb3e0-f3d5-11e7-9c95-fc8f623b5c79.png)
![tb-gateway-yml](https://user-images.githubusercontent.com/29800826/34649459-d1b80338-f3d5-11e7-886d-e16447d293f5.PNG)

I ran with different scenarios of MQTT broker running on Windows 10 or Raspberry Pi 3 and choosing between two (2) Gateways running on Windows 10 or Centos 7.

* * * *  Happy with Platform Configuration flexibility * * * * *

**Comment by ChSM56** [comment]
SOLVED:
MQTT Configuration in THINGSBOARD Platform with 3rd party MQTT broker being MOSQUITTO or HiveMQ and running on Centos 7 or Windows 10 systems.

**Comment by ChSM56** [comment]
@zhw618 , @ashvayka :

Please confirm the behaviour Thingsboard -

I have the platform running on local installation (CENTOS 7 ) and No TB-Gateway is running.

However, when a device is sending data, I notice the platform processing to show Gateway is Live 

Image to show tb-gateway (CENTOS 7) was shutdown -

![image](https://user-images.githubusercontent.com/29800826/34650603-bd9ee466-f3e9-11e7-81a2-fc71349d91a8.png)

Windows 10 - TB-GATEWAY is NOT RUNNING.
![image](https://user-images.githubusercontent.com/29800826/34650609-d8a5bbe0-f3e9-11e7-8dde-f826757cf1af.png)

However, Raspberry is able to send data to MQTT Broker running on Windows 10 system and THINGSBOARD platform is processing for Gateway status but not for the Device.

![image](https://user-images.githubusercontent.com/29800826/34650626-2316f22a-f3ea-11e7-9539-bdb9255411fa.png)

![image](https://user-images.githubusercontent.com/29800826/34650641-5cd2069e-f3ea-11e7-930e-f5df7270d40b.png)


Please help me to understand.

**Comment by ChSM56** [comment]
@zhw618 :
when you change .yml file from port 1883 to 1882，
the thingsboard will start a embedded mqtt broker using port 1882. 

What does this service on 1882 do?
Does it record the gateway that the device connects with ACCESS_TOKEN, though is NOT running records as LIVE?

Thanks for your clarification.

**Comment by zhw618** [comment]
I didn't  ever use  tb-gateway.

I only use  Thingsboard,  without any other third-party  mqtt broker ( mosquitto, hiveMQ, etc.)
just only installed Thingsboard.

when Thingsboard is installed correctly, 
you can use any  mqtt client to connect tb, which  acts as a  mqtt broker.
The *.yml file, configures the embedded mqtt broker.

the default port of  thingsboard's mqtt broker, is  1883,
if  you change the port to 1882, you device should connect thingboards's mqtt broker via 1882 port.
